jupyter_collaboration/rooms.py

```diff
@@ -234,12 +234,11 @@ async def _maybe_save_document(self, saving_document: asyncio.Task | None) -> No
         if saving_document is not None and not saving_document.done():
             # the document is being saved, cancel that
             saving_document.cancel()
```

Comment:
Is there a risk of corruption when cancelling a previous save? 

If the previous save if in the middle of a I/O action, would cancelling this task corrupt the file on disk?

Reply:
No because the saving task is [shielded](https://github.com/jupyterlab/jupyter-collaboration/blob/d4ec54a78ee60c364805643f72019852f62ac516/jupyter_collaboration/loaders.py#L145).

Reply:
Thanks for pointing me to this line. I'm still catching up on this codebase :)

Reply:
@Zsailer To clarify: there are three tasks that get run (most recent call last):

1. `DocumentRoom._maybe_save_document()`
2. `FileLoader.maybe_save_content()`
3. `FileLoader._save_content()`

(3) is the task writing to disk, and is shielded by (2) after it is created:

```
task = asyncio.create_task(self._save_content(model, done_saving))
try:
    await asyncio.shield(task)
except asyncio.CancelledError:
    pass
```

So in this context, (2) is a parent task of (3). When (2) runs (3) with `await asyncio.shield(task)`, this means that even if (1) or (2) get cancelled, the body of (3) will still proceed as expected.